pkg/gardenlet/controller/bastion/reconciler.go

```diff
@@ -46,49 +49,47 @@ const (
 	// finalizerName is the Kubernetes finalizerName that is used to control the cleanup of
 	// Bastion resources in the seed cluster.
 	finalizerName = gardencorev1alpha1.GardenerName
-
-	defaultTimeout         = 30 * time.Second
-	defaultSevereThreshold = 15 * time.Second
-	defaultInterval        = 5 * time.Second
 )
 
-// reconciler implements the reconcile.Reconcile interface for bastion reconciliation.
-type reconciler struct {
-	gardenClient client.Client
-	seedClient   client.Client
-	config       *config.GardenletConfiguration
-}
+var (
+	// DefaultTimeout is the default timeout value. Exposed for tests.
+	DefaultTimeout = pointer.Duration(30 * time.Second)
+	// DefaultInterval is the default interval value. Exposed for tests.
+	DefaultInterval = pointer.Duration(5 * time.Second)
+	// DefaultSevereThreshold is the default severe threshold value. Exposed for tests.
+	DefaultSevereThreshold = pointer.Duration(15 * time.Second)
+)
 
-// newReconciler returns the new bastion reconciler.
-func newReconciler(gardenClient, seedClient client.Client, config *config.GardenletConfiguration) reconcile.Reconciler {
-	return &reconciler{
-		gardenClient: gardenClient,
-		seedClient:   seedClient,
-		config:       config,
-	}
+// Reconciler reconciles Bastion and deploys them into the seed cluster.
+type Reconciler struct {
+	GardenClient client.Client
+	SeedClient   client.Client
+	Config       config.GardenletConfiguration
+	Clock        clock.Clock
 }
 
-func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+// Reconcile reconciles Bastion and deploys them into the seed cluster.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
 	bastion := &operationsv1alpha1.Bastion{}
-	if err := r.gardenClient.Get(ctx, request.NamespacedName, bastion); err != nil {
+	if err := r.GardenClient.Get(ctx, request.NamespacedName, bastion); err != nil {
 		if apierrors.IsNotFound(err) {
 			log.V(1).Info("Object is gone, stop reconciling")
 			return reconcile.Result{}, nil
 		}
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
-	if !IsBastionManagedByThisGardenlet(bastion, r.config) {
+	if !IsBastionManagedByThisGardenlet(bastion, &r.Config) {
```

Comment:
I don't think we still need this check when we centrally restrict the cache, do we?